### PR TITLE
[Doc] Explain the limitation of ray client for certain use case

### PR DESCRIPTION
**Pull request by scottsun94**
## Why are these changes needed?

- Ray Client has certain limitations when running ml workloads like tuning. We should be explicit about it
- remove "interactive development" in ray client page title because it feels like it is recommended for interactive development.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by scottsun94** [comment]
@richardliaw Does it capture the ray client limitation properly?

**Review by ckw017 — APPROVED** [review]

File: doc/source/cluster/running-applications/job-submission/ray-client.rst [hunk 1]
```diff
@@ -1,6 +1,6 @@
 .. _ray-client-ref:
 
-Ray Client: Interactive Development
+Ray Client
 ===================================
```
Comment (richardliaw) [inline doc/source/cluster/running-applications/job-submission/ray-client.rst:4]:
```suggestion
Ray Client
==========
```

File: doc/source/cluster/running-applications/job-submission/ray-client.rst [hunk 2]
```diff
@@ -26,9 +26,13 @@ By changing ``ray.init()`` to ``ray.init("ray://<head_node_host>:<port>")``, you
    do_work.remote(2)
    #....
 
+
 When to use Ray Client
 ----------------------
 
+.. note::
+   Ray Client has certain limitations and may not work as expected when using Ray for ML workloads that require sophisticated network communication. Use :ref:`Ray Jobs API<jobs-overview>` for interactive development on ML projects.
```
Comment (richardliaw) [inline doc/source/cluster/running-applications/job-submission/ray-client.rst:34]:
```suggestion
   Ray Client has architectural limitations and may not work as expected when using Ray for ML workloads -- for example, when used with Ray Tune or Ray Train. Use :ref:`Ray Jobs API<jobs-overview>` for interactive development on ML projects.
```

**Review by richardliaw — APPROVED** [review]

File: doc/source/cluster/running-applications/job-submission/ray-client.rst [hunk 3]
```diff
@@ -26,9 +26,13 @@ By changing ``ray.init()`` to ``ray.init("ray://<head_node_host>:<port>")``, you
    do_work.remote(2)
    #....
 
+
 When to use Ray Client
 ----------------------
 
+.. note::
+   Ray Client has architectural limitations and may not work as expected when using Ray for ML workloads -- for example, when used with Ray Tune or Ray Train. Use :ref:`Ray Jobs API<jobs-overview>` for interactive development on ML projects.
```
Comment (angelinalg) [inline doc/source/cluster/running-applications/job-submission/ray-client.rst:34]:
```suggestion
   Ray Client has architectural limitations and may not work as expected when using Ray for ML workloads (like Ray Tune or Ray Train). Use :ref:`Ray Jobs API<jobs-overview>` for interactive development on ML projects.
```

**Review by angelinalg — APPROVED** [review]

**Comment by scottsun94** [comment]
Not sure why it asked everyone for review. Sorry about the spam..

**Pull request merged**